include/newton_argument.h

```diff
@@ -0,0 +1,59 @@
+#ifndef newton_argument_h
+#define newton_argument_h
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_view.h>
+
+using namespace dealii;
+
+#ifdef DEAL_II_WITH_TRILINOS
```

Comment:
In this class you never use anything from Trilinos. You only declare an interface, correct? Then why exclude it if one does not have trilinos? Isn't this the same interface (more or less) of the ida solver?